### PR TITLE
HTTP Api to drop collection

### PR DESCRIPTION
**Pull request by ahsank**

### CODE REVIEW

File: src/skvhttpclient/src/skvhttp/client/SKVClient.cpp [hunk 1]
```diff
@@ -39,6 +39,15 @@ boost::future<Response<>> Client::createCollection(dto::CollectionMetadata metad
     return _HTTPClient.POST<dto::CollectionCreateRequest>("/api/CreateCollection", std::move(request));
 }
 
+boost::future<Response<>> Client::dropCollection(const String& collectionName) {
+    dto::CollectionDropRequest request{
+        .name = std::move(collectionName)
```
Comment (iafuture) [inline src/skvhttpclient/src/skvhttp/client/SKVClient.cpp:44]: A little surprised this compiles without a warning, but you should not move a const&. Pass by copy instead

File: src/skvhttpclient/test/ClientIntegrationTest.cpp [hunk 2]
```diff
@@ -526,6 +526,21 @@ void testReadWriteConflict() {
   }
 }
 
+void testDropCollection() {
```
Comment (iafuture) [inline src/skvhttpclient/test/ClientIntegrationTest.cpp:529]: Please also add a test to validate that we can create a new collection with the same name after the old one is dropped

**Review by iafuture — CHANGES_REQUESTED** [review]

File: src/skvhttpclient/src/skvhttp/client/SKVClient.cpp (hunk 1 above)
Comment (ahsank) [reply src/skvhttpclient/src/skvhttp/client/SKVClient.cpp:44]: Done

File: src/skvhttpclient/test/ClientIntegrationTest.cpp (hunk 2 above)
Comment (ahsank) [reply src/skvhttpclient/test/ClientIntegrationTest.cpp:529]: Done

**Review by jfunston — APPROVED** [review]

**Pull request merged**